ts/config.ts

```diff
@@ -163,21 +177,53 @@ export function load_secondary_theme_path(): string {
   return path_to_theme;
 }
 
+function validate_config(
+  decoder: io_ts.Type<typeof obj>,
+  obj: unknown
+): string {
+  const onLeft = (errors: io_ts.Errors): string => {
+    let message = "";
+    const fm_errors: { key: string; value: string; type: string }[] = [];
+    for (let i = 0; i < errors.length; i++) {
+      const context = errors[i].context;
+      const type = context[context.length - 1].type.name;
+      const key = context[context.length - 2].key;
+      const value = errors[i].value;
+      const can_color = process.stdout.isTTY && process.stdout.hasColors();
+      const fm_value = util.inspect(value, { colors: can_color });
+
+      const ind = fm_errors.findIndex((e) => e.key === key);
+      if (ind == -1) {
+        fm_errors.push({ key, value: fm_value, type });
+      } else {
+        fm_errors[ind].type += "|" + type;
+      }
+    }
+    for (const err of fm_errors) {
+      message += `{ ${err.key}: ${err.value} } couldn't be validated as (${err.type})\n`;
+    }
+    return message;
+  };
+  const onRight = (): string => "";
+  return pipe(decoder.decode(obj), fold(onLeft, onRight));
+}
+
 export function load_theme_config(): void {
   if (!theme_dir) load_theme_dir();
   const path_to_theme_config = path.join(theme_dir, "index.yml");
+  let error_message = "";
   try {
     const file = fs.readFileSync(path_to_theme_config, "utf-8");
-    const theme_config = yaml.load(file) as theme_config;
+    const theme_config = yaml.load(file);
 
-    if (!theme_config.primary_html.endsWith(".html"))
-      theme_config.primary_html = "index.html";
-    if (!theme_config.secondary_html.endsWith(".html"))
-      theme_config.secondary_html = "";
+    if (!THEME_CONFIG.is(theme_config)) {
+      error_message = validate_config(THEME_CONFIG, theme_config);
```

Comment:
Couple of things here to note regarding the decoding here...

* a value can fail `.is()` and succeed `.decode()` (for example decode can handle missing properties if that property is allowed to be undefined, because it returns its own value). As a result, `decode()` isn't guaranteed to produce errors that caused `.is()` to fail. So it's best to just simply call `.decode()` if we want to produce errors.
* you can just include the error message in the construction of the `Error()` rather than having a variable that you assign to sometimes.
* This code could be cleaned up a bit more if `validate_config` threw an error directly, or returned the valid value.